### PR TITLE
sql: make privilege order deterministic in error message

### LINKED ISSUE #118586: roachtest: pg_regress failed
roachtest.pg_regress [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13776729?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13776729?buildTab=artifacts#/pg_regress) on master @ [cc4fdffa8532d16544c48ef036689763f737dc6b](https://github.com/cockroachdb/cockroach/commits/cc4fdffa8532d16544c48ef036689763f737dc6b):

```
++github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:250: makeOptimizerPlan()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2295: makeExecPlan()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1762: dispatchToExecutionEngine()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1125: execStmtInOpenState()
 +github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:146: func1()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:3397: execWithProfiling()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:3427: execWithProfiling()
 +github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:145: execStmt()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2221: func1()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2226: execCmd()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2143: run()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:951: ServeConn()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2227: func1()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2232: execCmd()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2149: run()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:957: ServeConn()
 +github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:247: processCommands()
 +github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:1001: func3()
 +
@@ -275090,18 +274951,18 @@
 +github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:290: buildStmtAtRootWithScope()
 +github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:271: buildStmtAtRoot()
 +github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:250: Build()
-+github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:588: buildExecMemo()
-+github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:245: makeOptimizerPlan()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2276: makeExecPlan()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1744: dispatchToExecutionEngine()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1108: execStmtInOpenState()
++github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:593: buildExecMemo()
++github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:250: makeOptimizerPlan()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2295: makeExecPlan()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1762: dispatchToExecutionEngine()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1125: execStmtInOpenState()
 +github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:146: func1()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:3397: execWithProfiling()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:3427: execWithProfiling()
 +github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:145: execStmt()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2221: func1()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2226: execCmd()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2143: run()
-+github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:951: ServeConn()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2227: func1()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2232: execCmd()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2149: run()
++github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:957: ServeConn()
 +github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:247: processCommands()
 +github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:1001: func3()
 +
If the diff is expected, copy regression.diffs from the test artifacts to pkg/cmd/roachtest/testdata/regression.diffs
test artifacts and logs in: /artifacts/pg_regress/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13776729/pg-regress/1706809684833/1706810840752)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pg_regress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-35800

### PR DESCRIPTION
**Pull request by rharding6373**
Before this change, if a user did not have correct privileges to run a command, the error message could print the missing privileges in any order. This commit makes this order deterministic to reduce test flakiness.

Epic: none
Fixes: #118586

Release note: None

### CODE REVIEW

**Review by michae2 — APPROVED** [review]
It's the little things... :lgtm:

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/118889#-:-Nq3hQou9luHz_r2xg2c:bdwhtju)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rharding6373)

**Comment by rharding6373** [comment]
TFTR! I missed a couple tests in CCL, so I'll wait for CI to be green before merging.

I'm uncertain about how this has not been a problem in our logic tests before, but didn't dig around.

**Comment by rharding6373** [comment]
The test failures are unrelated. 

bors r+

**Pull request merged**
